### PR TITLE
Explicitly define ignored by UNT_MANHOLE.

### PR DESCRIPTION
**Pull request by MrKeiKun**
### Pull Request Prelude

- [x] I have followed [proper Hercules code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed
Explicitly define the targets that should be ignored by UNT_MANHOLE.

**Issues addressed:** #2964 

[cont]: https://github.com/HerculesWS/Hercules/blob/master/CONTRIBUTING.md
[code]: https://github.com/HerculesWS/Hercules/wiki/Coding-Style

### CODE REVIEW

File: src/map/skill.c [hunk 1]
```diff
@@ -13385,11 +13385,10 @@ static int skill_unit_onplace_timer(struct skill_unit *src, struct block_list *b
 		case UNT_MANHOLE:
 			if (sg->val2 == 0 && tsc && (sg->unit_id == UNT_ANKLESNARE || bl->id != sg->src_id)) {
 				int sec = skill->get_time2(sg->skill_id, sg->skill_lv);
-				if (bl->type == BL_MOB && map->list[bl->m].flag.gvg_castle == 0) {
-					struct mob_data *md = BL_UCAST(BL_MOB, bl);
-					if (sg->unit_id == UNT_MANHOLE && (md->class_ == MOBID_EMPELIUM || md->class_ == MOBID_BARRICADE || md->class_ == MOBID_S_EMPEL_1 || md->class_ == MOBID_S_EMPEL_2)) {
-						// Do nothing...
-					}
+				struct mob_data *md = BL_UCAST(BL_MOB, bl);
+				if (sg->unit_id == UNT_MANHOLE &&
```
Comment (dastgirp) [inline src/map/skill.c:13389]: md can be NULL here, as it's directly casted without checking the type.

Comment (MrKeiKun) [reply src/map/skill.c:13389]: yeah thats another problem i saw.. either i have to redo the whole condition above it

File: src/map/skill.c [hunk 2]
```diff
@@ -13385,10 +13385,10 @@ static int skill_unit_onplace_timer(struct skill_unit *src, struct block_list *b
 		case UNT_MANHOLE:
 			if (sg->val2 == 0 && tsc && (sg->unit_id == UNT_ANKLESNARE || bl->id != sg->src_id)) {
 				int sec = skill->get_time2(sg->skill_id, sg->skill_lv);
-				if (bl->type == BL_MOB && map->list[bl->m].flag.gvg_castle == 0) {
+				if (sg->unit_id == UNT_MANHOLE && bl->type == BL_MOB) {
```
Comment (MrKeiKun) [inline src/map/skill.c:13388]: this is bad. this works to all monsters inside.

File: src/map/skill.c [hunk 3]
```diff
@@ -13385,10 +13385,10 @@ static int skill_unit_onplace_timer(struct skill_unit *src, struct block_list *b
...
 					struct mob_data *md = BL_UCAST(BL_MOB, bl);
-					if (sg->unit_id == UNT_MANHOLE && (md->class_ == MOBID_EMPELIUM || md->class_ == MOBID_BARRICADE || md->class_ == MOBID_S_EMPEL_1 || md->class_ == MOBID_S_EMPEL_2)) {
-						// Do nothing...
+					if (md->class_ == MOBID_EMPELIUM || md->class_ == MOBID_BARRICADE || md->class_ == MOBID_S_EMPEL_1 || md->class_ == MOBID_S_EMPEL_2) {
+						// Do nothing if target are the specified monsters on manhole.
```
Comment (olsonpm) [inline src/map/skill.c:13391]: what is the purpose of this inner block ?  it doesn't seem to cause any side effects so there seems to be no pint in the inner if

File: src/map/skill.c [hunk 4]
```diff
@@ -13385,11 +13385,10 @@ static int skill_unit_onplace_timer(struct skill_unit *src, struct block_list *b
 		case UNT_MANHOLE:
 			if (sg->val2 == 0 && tsc && (sg->unit_id == UNT_ANKLESNARE || bl->id != sg->src_id)) {
 				int sec = skill->get_time2(sg->skill_id, sg->skill_lv);
-				if (bl->type == BL_MOB && map->list[bl->m].flag.gvg_castle == 0) {
-					struct mob_data *md = BL_UCAST(BL_MOB, bl);
-					if (sg->unit_id == UNT_MANHOLE && (md->class_ == MOBID_EMPELIUM || md->class_ == MOBID_BARRICADE || md->class_ == MOBID_S_EMPEL_1 || md->class_ == MOBID_S_EMPEL_2)) {
-						// Do nothing...
-					}
+				if (sg->unit_id == UNT_MANHOLE && bl->type == BL_MOB &&
+					(BL_UCCAST(BL_MOB, bl)->class_ == MOBID_EMPELIUM || BL_UCCAST(BL_MOB, bl)->class_ == MOBID_BARRICADE || BL_UCCAST(BL_MOB, bl)->class_ == MOBID_S_EMPEL_1 || BL_UCCAST(BL_MOB, bl)->class_ == MOBID_S_EMPEL_2)
```
Comment (MrKeiKun) [inline src/map/skill.c:13389]: @dastgirp what do you think.. i'm not a fan of it...

**Comment by olsonpm** [comment]
can you please answer my questions?  i'd like to help and your code is both not making sense and causing bugs

**Review by MishimaHaruna — APPROVED** [review]

**Pull request merged**